ui/app/components/tools/wrap.js

```diff
@@ -46,13 +63,13 @@ export default class ToolsWrap extends Component {
     codemirror.performLint();
     const hasErrors = codemirror?.state.lint.marked?.length > 0;
     this.buttonDisabled = hasErrors;
-    if (!hasErrors) this.wrapData = val;
+    if (!hasErrors) this.wrapData = JSON.parse(val);
```

Comment:
It looks like we get into a weird state if there are linting errors and the user toggles the `JSON` button (go to the key/value view and then back to the json editor. the JSON "resets", though it still shows there's a linting issue and the wrap button is still enabled. If a user toggles to the other view, perhaps we should clear out the linting errors as well.